### PR TITLE
CLI - install multiple pkgs from folder

### LINKED ISSUE #14398: [Feature request] Install multiple PKG from command line
### Quick summary

Right now, the option, --installpkg in the command line only allow to install a single pkg
It would be nice if we could put a folder as argument an get the UI that allow to install multiple pkg

image

### Details

_No response_

### PR DESCRIPTION
**Pull request by nkarl7**
Fixes #14398

### CODE REVIEW

**Comment by Megamouse** [comment]
This is not really preferable.
You're supposed to install these packages in a specific order.
Having this more or less arbitrary based on dir file order will only lead to unwanted issues.

**Comment by Megamouse** [comment]
Ah sorry. I didn't read the code wrong. I'll better look at it sober tomorrow 😆

File: rpcs3/rpcs3qt/main_window.cpp [hunk 1]
```diff
@@ -743,75 +743,93 @@ bool main_window::InstallPackages(QStringList file_paths, bool from_boot)
...
-	if (file_paths.count() == 1 && file_paths.front().endsWith(".pkg", Qt::CaseInsensitive))
+	if (file_paths.count() == 1)
 	{
 		const QString file_path = file_paths.front();
 		const QFileInfo file_info(file_path);
 
-		compat::package_info info = game_compatibility::GetPkgInfo(file_path, m_game_list_frame ? m_game_list_frame->GetGameCompatibility() : nullptr);
-
-		if (!info.is_valid)
+		// #14398 - [Feature request] Install multiple PKG from command line
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/main_window.cpp:751]: This comment is unnecessary and adds nothing to the code.

File: rpcs3/rpcs3qt/main_window.cpp [hunk 2]
```diff
@@ -743,75 +743,93 @@ bool main_window::InstallPackages(QStringList file_paths, bool from_boot)
...
+		if (file_info.isDir())
 		{
-			QMessageBox::warning(this, tr("Invalid package!"), tr("The selected package is invalid!\n\nPath:\n%0").arg(file_path));
-			return false;
-		}
+			const QDir dir(file_path);
+			const auto dir_file_infos = dir.entryInfoList(QDir::Files);
+			const auto dir_file_infos_count = dir_file_infos.count();
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/main_window.cpp:756]:
```suggestion
			gui_log.notice("PKG: Trying to install packages from dir: '%s'", file_path);

			const QDir dir(file_path);
			const auto dir_file_infos = dir.entryInfoList({"*.pkg"}, QDir::Files);

			if (dir_file_infos.empty())
			{
				gui_log.notice("PKG: Could not find any packages in dir: '%s'", file_path);
				return true;
			}

			const auto dir_file_infos_count = dir_file_infos.count();
```

Comment (nkarl7) [reply rpcs3/rpcs3qt/main_window.cpp:756]: Wouldn't filtering by .pkg here potentially stop raps and edats from getting installed afterwards? I'm not sure what lines 837 onwards do though, to be perfectly honest.

Comment (Megamouse) [reply rpcs3/rpcs3qt/main_window.cpp:756]: ah, right, without the filter then.

File: rpcs3/rpcs3qt/main_window.cpp [hunk 3]
```diff
@@ -743,75 +743,102 @@ bool main_window::InstallPackages(QStringList file_paths, bool from_boot)
...
-		}
 
-		if (!info.title_id.isEmpty())
-		{
-			info.title_id = tr("\n%0", "Block for Title ID").arg(info.title_id);
+			return InstallPackages(dir_file_paths, from_boot);
 		}
-
-		if (!info.version.isEmpty())
+		else if (file_info.suffix().compare("pkg", Qt::CaseInsensitive) == 0)
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/main_window.cpp:775]: else is not needed
```suggestion

		if (file_info.suffix().compare("pkg", Qt::CaseInsensitive) == 0)
```

File: rpcs3/rpcs3qt/main_window.cpp (hunk 1 above)
Comment (nkarl7) [reply rpcs3/rpcs3qt/main_window.cpp]: Removed

**Pull request merged**